pkg/vpcmodel/nodesExplainability.go

```diff
@@ -25,22 +36,113 @@ func (c *VPCConfig) getVsiNode(name string) Node {
 	return nil
 }
 
-// ExplainConnectivity todo: this will not be needed here once we connect explanbility to the cli
-// todo: add support of external network
-func (c *VPCConfig) ExplainConnectivity(srcName, dstName string) (explanation string, err error) {
-	src := c.getVsiNode(srcName)
-	if src == nil {
-		return "", fmt.Errorf("src %v does not represent a VSI", srcName)
+// given input cidr, gets (disjoint) external nodes I s.t.:
+//  1. The union of these nodes is the cidr
+//  2. Let i be a node in I and n be a node in VPCConfig.
+//     i and n are either disjoint or i is contained in n
+//     Note that the vpconfig nodes were chosen w.r.t. connectivity rules (SG and NACL)
+//     s.t. each node either fully belongs to a rule or is disjoint to it.
+//     to get nodes I as above:
+//  1. Calculate the IP blocks of the nodes N
+//  2. Calculate from N and the cidr block, disjoint IP blocks
+//  3. Return the nodes created from each block from 2 contained in the input cidr
+func (c *VPCConfig) getCidrExternalNodes(cidr string) (cidrNodes []Node, err error) {
+	cidrsIPBlock := common.NewIPBlockFromCidr(cidr)
+	if cidrsIPBlock == nil { // string cidr does not represent a legal cidr
+		return nil, nil
+	}
+	// 1.
+	vpcConfigNodesExternalBlock := []*common.IPBlock{}
+	for _, node := range c.Nodes {
+		if node.IsInternal() {
+			continue
+		}
+		thisNodeBlock := common.NewIPBlockFromCidr(node.Cidr())
+		vpcConfigNodesExternalBlock = append(vpcConfigNodesExternalBlock, thisNodeBlock)
 	}
-	dst := c.getVsiNode(dstName)
-	if dst == nil {
-		return "", fmt.Errorf("dst %v does not represent a VSI", dstName)
+	// 2.
+	disjointBlocks := common.DisjointIPBlocks([]*common.IPBlock{cidrsIPBlock}, vpcConfigNodesExternalBlock)
+	// 3.
+	cidrNodes = make([]Node, 0)
+	for _, block := range disjointBlocks {
+		if block.ContainedIn(cidrsIPBlock) {
+			node, err1 := newExternalNode(true, block)
+			if err1 != nil {
+				return nil, err1
+			}
+			cidrNodes = append(cidrNodes, node)
+		}
 	}
-	rulesOfConnection, err1 := c.GetRulesOfConnection(src, dst)
+	return cidrNodes, nil
+}
+
+// given a string or a vsi or a cidr returns the corresponding node(s)
+func (c *VPCConfig) getNodesFromInput(cidrOrName string) ([]Node, error) {
+	if vsi := c.getVsiNode(cidrOrName); vsi != nil {
```

Comment:
Consider as todo to later support input vsi given as CRN/ID or as IP address of network interface, and not only by name

Reply:
Sure. Lets talk shortly and prioritize. If this is part of the first release I'll do it now.